### PR TITLE
feat: Vercel marketplace OIDC

### PR DESCRIPTION
**Pull request by MildTomato**
## What is the new behavior?

- adds Vercel marketplace provider

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: internal/api/provider/vercel_marketplace.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func (g vercelMarketplaceProvider) GetOAuthToken(code string) (*oauth2.Token, error) {
+	return g.Exchange(context.Background(), code)
+}
+
+func (g vercelMarketplaceProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*UserProvidedData, error) {
+	idToken := tok.Extra("id_token")
+	if tok.AccessToken == "" || idToken == nil {
+		return &UserProvidedData{}, nil
```
Comment (hf) [inline internal/api/provider/vercel_marketplace.go:65]: This should probably be an error, no? If there's no user data?

File: internal/conf/configuration.go [hunk 2]
```diff
@@ -322,6 +322,7 @@ type ProviderConfiguration struct {
 	SlackOIDC               OAuthProviderConfiguration     `json:"slack_oidc" envconfig:"SLACK_OIDC"`
 	Twitter                 OAuthProviderConfiguration     `json:"twitter"`
 	Twitch                  OAuthProviderConfiguration     `json:"twitch"`
+	VercelMarketplace       OAuthProviderConfiguration     `json:"vercel_marketplace" envconfig:"VERCEL_MARKETPLACE"`
```
Comment (hf) [inline internal/conf/configuration.go:325]:
```suggestion
	VercelMarketplace       OAuthProviderConfiguration     `json:"vercel_marketplace" split_words:"true"`
```

No need for a special envconfig. It should be configurable via `GOTRUE_EXTERNAL_VERCEL_MARKETPLACE...`

**Review by hf — APPROVED** [review]
Give the comments a consideration and merge whenever you want. :D

File: internal/api/provider/vercel_marketplace.go (hunk 1 above)
Comment (kamilogorek) [reply internal/api/provider/vercel_marketplace.go]: Followed error message format from `azure` implementation

**Comment by kamilogorek** [comment]
image

Well done @MildTomato! :D

**Comment by MildTomato** [comment]
> Well done @MildTomato! :D

lolol

**Pull request merged**